test/Chainweb/Test/SPV.hs

```diff
@@ -101,6 +118,140 @@ targetChain c srcBlock = do
+            -- Ingore all cases where a proof couldn't be created
+            & S.concat
+            & S.toList_
+
+        -- Confirm size of proofs
+        let (coef, r) = regress samples
+        step $ show r
+        step $ show coef
+        assertBool "proof size is not constant in the block height" (r > 0.8)
+
+        let (coef', r') = regressWithHeightDiff samples
+        step $ show r'
+        step $ show coef'
+        assertBool "proof size is not constant in the block height" (abs (coef' V.! 1) < 1)
+  where
+
+    -- Stream of all transactions in a block. Each returned item is
+    -- - block header
+    -- - block size (tx count)
+    -- - tx index
+    -- - tx
+    --
+    getPayloads
+        :: PayloadCas cas
+        => CutDb cas
+        -> BlockHeader
+        -> S.Stream (Of (BlockHeader, Int, Int, Transaction)) IO ()
+    getPayloads cutDb h = do
+        pay <- liftIO $ casLookupM (view cutDbPayloadCas cutDb) (_blockPayloadHash h)
+        let n = length $ _payloadWithOutputsTransactions pay
+        S.each (zip [0..] $ fmap fst $ toList $ _payloadWithOutputsTransactions pay)
+            & S.map (\(b,c) -> (h,n,b,c))
+
+    -- Given
+    -- - block header,
+    -- - tx count of block,
+    -- - tx index,
+    -- - tx, and
+    -- - target chain,
+    --
+    -- creates and verifies SPV proofs and returns list of
+    -- - tx proof length
+    -- - block size (tx count)
+    -- - block height difference
+    -- - distance between source chain and target chain
+    -- - size of tx
+    --
+    go
+        :: PayloadCas cas
+        => CutDb cas
+        -> (BlockHeader, Int, Int, Transaction, ChainId)
+        -> IO (Maybe [Double])
+    go cutDb (h, n, txIx, tx, trgChain) = do
+
+        let inner = do
+                -- create inclusion proof for transaction
+                proof <- createTransactionProof cutDb trgChain
+                    (_chainId h) -- source chain
+                    (_blockHeight h) -- source block height
+                    txIx -- transaction index
+                subj <- verifyTransactionProof cutDb proof
+                assertEqual "transaction proof subject matches transaction" tx subj
+
+                -- return (proof size, block size, height, distance, tx size)
+                return
+                    [ int $ BL.length $ encode proof
+                    , int n
+                    , int $ _blockHeight h
+                    , int $ distance cutDb h trgChain
+                    , int $ B.length (_transactionBytes tx)
+                    ]
+
+        isReachable <- reachable cutDb h trgChain
+        try inner >>= \case
+            Right x -> do
+                assertBool "SPV proof creation succeeded although target chain is not reachable" isReachable
+                return (Just x)
+            Left SpvExceptionTargetNotReachable{} -> do
+                assertBool "SPV proof creation failed although target chain is reachable" (not isReachable)
```

Comment:
Is this a success case? It's not clear why this would fail the test

Reply:
also, again, if there's a chance a maintainer can screw this up, please have the `assertXXX` message output relevant data

Reply:
The exception may be thrown in the tested code, so it's part of the tested `createTransactionProof` API. The assertion checks that the exception is only thrown when it is expected.

I can add the height difference and chain distance data to the failure message, which determines if a proof exists or not.